c/build.c

```diff
@@ -146,47 +159,105 @@ void prepend_to_data_file()
         fprintf(data_file, "%s", line);
     }
 
-    fprintf(data_file, "\n#endif\n");
-
     fclose(temp_file);
     if (line)
         free(line);
     remove(TEMP_FILE);
     fclose(data_file);
 }
 
+/**
+ * Add `get_node_by_path` function definition.
+ * @returns This function do not return anything.
+ */
+void append_to_data_file()
+{
+    FILE *data_file = fopen(DATA_FILE, "a");
+    if (data_file == NULL)
+        exit(EXIT_FAILURE);
+
+    // Add init function to allocate memory for necessary heap allocated data
+    fprintf(data_file, "\nvoid __kuliya_init()\n{\n");
+    for (size_t i = 0; i < kuliya_with_terms_idx; ++i)
+    {
+        __s_kuliya_schema schema = kuliyas_with_terms[i];
+        fprintf(data_file, "\t%s.terms = malloc(sizeof(kuliya_terms));\n", schema.__varname);
+        fprintf(data_file, "\t%s.terms->per_year = %d;\n", schema.__varname, schema.terms.per_year);
+        fprintf(data_file, "\t%s.terms->number_of_slots = %zu;\n", schema.__varname, schema.__slots_length);
+        fprintf(data_file, "\t%s.terms->slots = malloc(%zu * sizeof(int));\n", schema.__varname, schema.__slots_length);
+        for (size_t j = 0; j < schema.__slots_length; ++j)
+        {
+            fprintf(data_file, "\t%s.terms->slots[%zu] = %d;\n", schema.__varname, j, schema.terms.slots[j]);
+        }
+    }
+    fprintf(data_file, "}\n");
+
+    // Add deinit function to free heap allocated data
+    fprintf(data_file, "\nvoid __kuliya_deinit()\n{\n");
+    for (size_t i = 0; i < kuliya_with_terms_idx; ++i)
+    {
+        __s_kuliya_schema schema = kuliyas_with_terms[i];
+        fprintf(data_file, "\tfree(%s.terms->slots);\n", schema.__varname);
+        fprintf(data_file, "\tfree(%s.terms);\n", schema.__varname);
+    }
+    fprintf(data_file, "}\n");
+
+    // Prepare get node by path API
+    fprintf(data_file, "\nkuliya_schema *__get_node_by_path(const char* path)\n{");
```

Comment:
```suggestion
    fprintf(data_file, "\nkuliya_schema *__get_node_by_path(const char *path)\n{");
```

the C/C++ extension in VSCode keeps on formatting it this way, i guess this is the correct formatting then?

Reply:
right